readthedocs/api/v2/views/model_views.py

```diff
@@ -300,14 +313,36 @@ def reset(self, request, **kwargs):
         instance.reset()
         return Response(status=status.HTTP_204_NO_CONTENT)
 
+    def get_queryset(self):
```

Comment:
Do you mean the queryset itself or getting the api key? The querysets may look the same, but each queryset is different, since each resource is related to a project differently (`pk=project.pk`, `project=project`, `build_project=`...). If it's about getting the key, not sure if a mixin will save much code (but it can be moved to the `UserSelectViewSet` class, and have a method `get_queryset_for_api_key(api_key)` or similar).

Reply:
Hrm yea, I totally missed it was subtly changing each query.... 🙃